kratos/utilities/parallel_utilities.h

```diff
@@ -206,7 +206,7 @@ class BlockPartition
         KRATOS_PREPARE_CATCH_THREAD_EXCEPTION
 
         TReducer global_reducer;
-        #pragma omp parallel for
+        #pragma omp parallel for schedule(runtime)
```

Comment:
@loumalouomega as i am telling, take a look to line 154. It does not make sense to change this unless we change what happens there.

also to my understanding the runtime behaviour has potentially a very high overhead due to the need of making a syscall to fetch an env variable. 

https://stackoverflow.com/questions/7460552/reading-environment-variables-is-slow-operation/7460612#7460612

not sure if that matters...but at least we need to beware of this

Reply:
We should use the benchmark to check that it affects significantly

Reply:
The idea of the runtime is to give flexibility, if you prefer we can define it on compiling time...

Reply:
It is recommended by OMP itself:

https://www.openmp.org/wp-content/uploads/openmp-webinar-vanderPas-20210318.pdf

![image](https://github.com/user-attachments/assets/156006e4-1adf-45a8-9100-1d2213213741)

(And I found a master thesis saying that it doesn't penalize https://hpc.dmi.unibas.ch/wp-content/uploads/sites/87/2020/10/2019_akan_yilmaz_ma_thesisjune2019.pdf)

Reply:
@loumalouomega aside of the comments on the opportunity of using the OMP_SCHEDULE did u take a look at what i am writing? 

**we are doing "by hand" the chunking**. If we don't change that, it makes no sense to use a different scheduling, as everyone will be working on its chunk (as of now **we dot have more chunks than threads!**)

Reply:
Okay...let me think this...

Reply:
![image](https://github.com/user-attachments/assets/36c3ed08-8a92-47b0-940b-e1ce7df168f2)

Centrantly there is no effect (significative), I need to rethink this...

Reply:
In that case we may need to rethink the chunging (to be dependent of the CPU architecture)

Reply:
@RiccardoRossi what do you suggest exactly, because I has been studying this and our chunking conflicts with the OMP scheduling, and a priori the most efficient would be to let OMP to do the chunking. The problem is that with that we lose the parallel_utilities design and reduction utilities.